File: billing_back/billing_back/Models/BillPurchases.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using billing_back.Models;

namespace billing_back.Models
{
    public class BillPurchases
    {
        public Bill Bill { get; set; }
        public Purchase[] Purchases { get; set; }
    }
}

File: billing_presentation/src/app/core/models/BillPurchases.ts
import { Bill } from "../models/Bill";
import { Purchase } from "../models/Purchase";

export interface BillPurchases {

  bill: Bill;
  purchases: Purchase[];

}

export const BillPurchasesNull =
{
  bill: {
    IdBill: 0,
    IdClient: 0,
    PurchaseDate: new Date("01/01/1900"),
  },
  purchases:[
    {
      IdPurchase: 0,
      IdBill: 0,
      IdProduct: 0,
      ProductQuantity: 0,
      PurchaseValue: 0
    }
  ]
}
